From ec8ce4fb702f99128da391215671e4de99d3d5e5 Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <mian.kaukab@intel.com>
Date: Thu, 19 Jun 2014 15:48:11 +0200
Subject: [PATCH] usb: gadget: android: remove redundant unbind calls

Calling usb_remove_function in unbind calls results in redundant
f->disable and f->unbind calls as usb_remove_config takes care of
calling these.

Change-Id: Id79b0e44db2eac38c3cd30b5a64a71fd05ce27bb
Signed-off-by: Mian Yousaf Kaukab <mian.kaukab@intel.com>
---
 drivers/usb/gadget/android.c | 33 ---------------------------------
 1 file changed, 33 deletions(-)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index 8be314423f14..2ceb4593d800 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -329,16 +329,6 @@ err_usb_add_function:
 	return ret;
 }
 
-static void ffs_function_unbind_config(struct android_usb_function *f,
-				       struct usb_configuration *c)
-{
-	int i;
-	struct functionfs_config *config = f->config;
-
-	for (i = 0; i < config->instances_on; i++)
-		usb_remove_function(c, config->f_ffs[i]);
-}
-
 static ssize_t
 ffs_aliases_show(struct device *pdev, struct device_attribute *attr, char *buf)
 {
@@ -413,7 +403,6 @@ static struct android_usb_function ffs_function = {
 	.disable	= ffs_function_disable,
 	.cleanup	= ffs_function_cleanup,
 	.bind_config	= ffs_function_bind_config,
-	.unbind_config	= ffs_function_unbind_config,
 	.attributes	= ffs_function_attributes,
 };
 
@@ -544,16 +533,6 @@ err_usb_add_function:
 	return ret;
 }
 
-static void acm_function_unbind_config(struct android_usb_function *f,
-				       struct usb_configuration *c)
-{
-	int i;
-	struct acm_function_config *config = f->config;
-
-	for (i = 0; i < config->instances_on; i++)
-		usb_remove_function(c, config->f_acm[i]);
-}
-
 static ssize_t acm_instances_show(struct device *dev,
 		struct device_attribute *attr, char *buf)
 {
@@ -588,7 +567,6 @@ static struct android_usb_function acm_function = {
 	.init		= acm_function_init,
 	.cleanup	= acm_function_cleanup,
 	.bind_config	= acm_function_bind_config,
-	.unbind_config	= acm_function_unbind_config,
 	.attributes	= acm_function_attributes,
 };
 
@@ -928,16 +906,6 @@ err_usb_add_function:
 	return ret;
 }
 
-static void mass_storage_function_unbind_config(struct android_usb_function *f,
-					       struct usb_configuration *c)
-{
-	int i;
-	struct mass_storage_function_config *config = f->config;
-
-	for (i = 0; i < config->instances_on; i++)
-		usb_remove_function(c, config->f_ms[i]);
-}
-
 static ssize_t mass_storage_inquiry_show(struct device *dev,
 				struct device_attribute *attr, char *buf)
 {
@@ -974,7 +942,6 @@ static struct android_usb_function mass_storage_function = {
 	.init		= mass_storage_function_init,
 	.cleanup	= mass_storage_function_cleanup,
 	.bind_config	= mass_storage_function_bind_config,
-	.unbind_config	= mass_storage_function_unbind_config,
 	.attributes	= mass_storage_function_attributes,
 };
 
-- 
1.9.1

